Source/WebCore/dom/FullscreenManager.cpp

```diff
@@ -120,8 +122,11 @@ void FullscreenManager::requestFullscreenForElement(Ref<Element>&& element, RefP
     // If any of the following conditions are true, terminate these steps and queue a task to fire
     // an event named fullscreenerror with its bubbles attribute set to true on the context object's
     // node document:
-    if (!element->isHTMLElement() && !is<SVGSVGElement>(element) && !is<MathMLMathElement>(element)) {
-        handleError("Cannot request fullscreen on a non-HTML element."_s, EmitErrorEvent::Yes, WTFMove(element), WTFMove(promise), WTFMove(completionHandler));
+    if (!element->isHTMLElement() && !is<SVGSVGElement>(element)) {
+#if ENABLE(MATHML)
+        if (!is<MathMLMathElement>(element))
+#endif
+            handleError("Cannot request fullscreen on a non-HTML element."_s, EmitErrorEvent::Yes, WTFMove(element), WTFMove(promise), WTFMove(completionHandler));
         return;
```

Comment:
I looks to me like this changes behavior and early returns in cases where it previously didn't.

Reply:
Good catch, morning PRs are not the best for me 😛

Reply:
Personally I'd refactored to something like this:
```
static bool isElementAllowedToRequestFullscreen(const Element& element)
{
    if (is<HTMLElement>(element) || is<SVGElement>(element))
        return true;
#if ENABLE(MATHML)
    if (is<MathMLElement>(element))
        return true;
#endif
    return false;
}
```

Then
```
if (! isElementAllowedToRequestFullscreen) {
    handleError("Cannot request fullscreen on a non-HTML element."_s, EmitErrorEvent::Yes, WTFMove(element), WTFMove(promise), WTFMove(completionHandler));
    return;
}
```